google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -103,6 +103,7 @@ public com.google.api.services.storage.model.Bucket apply(BucketInfo bucketInfo)
   private final IamConfiguration iamConfiguration;
   private final String locationType;
   private final Logging logging;
+  private final CustomPlacementConfig customPlacementConfig;
```

Comment:
How does custom placement differ from `location` above?

Should we refactor away location to always use this multi-region type thing with a single element?

Reply:
Good catch, both `location` and `customPlacementConfig` will need to be specified for (custom) dual-regions. Here's an example:
- `location` = 'US'
- `customPlacementConfig` = 'US-EAST-1' and 'US-WEST-1'

Reply:
This should be made for the gRPC protos as well.

Reply:
@sydney-munro and @frankyn 

Given the information from Dan above: Do we want to introduce an actual higher level data type which can model the `location` vs `location + customPlacementConfig`, where our BucketInfo would then use this new type? Or, do we hope any possible error message from the backend helps folks figure out they need to be set together?

Reply:
I spoke with Dan offline and the backend should be handling this. I think the main reason to not do this is that using location does not necessarily mean you need customPlacementConfig.

Reply:
I was thinking of it from the other direction. If I'm using custom placement config I would guess that I choose either `location` or `custom placement config`, not choose either `location` or `location and custom placement config`.

Reply:
As I was working on coding this I do have a concern around this as a breaking change, do we want people that don't care about multiple regions that are using location to have to worry about changing their code just to be able to use the latest client?

Reply:
sync'd offline. we're going to go ahead with the initial proposal, deferring this new type to a later time if we need to.